src/olympia/files/utils.py

```diff
@@ -868,9 +869,13 @@ def check_xpi_info(xpi_info, addon=None, xpi_file=None, user=None):
             or DeniedGuid.objects.filter(guid=guid).exists()
         ):
             raise forms.ValidationError(gettext('Duplicate add-on ID found.'))
-    if len(xpi_info['version']) > 32:
+
+    version_field_max_length = Version.version.field.max_length
+    if len(xpi_info['version']) > version_field_max_length:
         raise forms.ValidationError(
-            gettext('Version numbers should have fewer than 32 characters.')
+            gettext(
+                'Version numbers should have fewer than {max_length} characters.'
+            ).format(max_length=version_field_max_length)
```

Comment:
Technically we allow up to `max_length`, so should add a `+1` here when formatting the message or tweak it.

Reply:
So the message was wrong for 12 years and no-one noticed...